k8s/upgrades/0.8.2-0.9.0/cstor/cstor-pool-update-082-090.yaml

```diff
@@ -0,0 +1,664 @@
+    - upgrade-cstor-pool-082-090-get-storagepool
+    # puts result og resource getting via above 3 runtasks
+    - upgrade-cstor-pool-082-090-patch-upgrade-result
+    # patches pool deployment with target version
+    - upgrade-cstor-pool-082-090-patch-deployment-image
+    # Waits until new pods have came up successfully
+    - upgrade-cstor-pool-082-090-patch-deployment-image-status
+    # checks whether pool containers in target version or not
+    - upgrade-cstor-pool-082-090-post-check-patch-deployment-image
+    # lists all replicasets of pool deployment
+    - upgrade-cstor-pool-082-090-patch-sp-version
+    - upgrade-cstor-pool-082-090-patch-sp-version-post-check
+    - upgrade-cstor-pool-082-090-patch-csp-version
+    - upgrade-cstor-pool-082-090-patch-csp-version-post-check
+    - upgrade-cstor-pool-082-090-patch-deployment-version
+    - upgrade-cstor-pool-082-090-patch-deployment-version-post-check
+    - upgrade-cstor-pool-082-090-list-replicaset
+    # list working replicaset and pod of pool deployment
+    - upgrade-cstor-pool-082-090-list-pod
+    - upgrade-cstor-pool-082-090-delete-replicaset
+  taskNamespace: default
+---
+# upgrade-cstor-pool-082-090-get-cstorpool gets details of CStorPool CR.
+# This runtask is for get object info which will use in some runtask.
+# This should always run.
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-cstor-pool-082-090-get-cstorpool
+  namespace: default
+spec:
+  meta: |
+    id: CStorPool
+    apiVersion: openebs.io/v1alpha1
+    kind: CStorPool
+    action: get
+    objectName: {{ .UpgradeItem.name }}
+    disable: false
+  post: |
+    {{- jsonpath .JsonResult "{.metadata.uid}" | trim | saveAs "CStorPool.uid" .TaskResult | noop -}}
+    {{- jsonpath .JsonResult "{.metadata.labels.openebs\\.io/version}" | trim | saveAs "CStorPool.labels" .TaskResult | noop -}}
+
+    {{- $message := printf "Successfully got details of CStorPool {%s}." .UpgradeItem.name -}}
+    {{- $status :="" }}
+    {{- if eq .TaskResult.CStorPool.labels "0.8.2" }}
+    {{- $status =.Config.successStatus.value }}
+    {{- else }}
+    {{- $status =.Config.failStatus.value }}
+    {{- end }}
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-get-cstorpool" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+    {{- if eq $status .Config.failStatus.value }}
+    {{- verifyErr "CStorPool version is not 0.8.2" true | saveAs "CStorPool.verifyErr" .TaskResult | noop -}} 
```

Comment:
Is it possible to have all the message defined in post at one place?
We can provide extra info like CSP name etc in these messages
e.g.
```yaml
{{- $successMsg := "" -}}
{{- $errorMsg := "" -}}
{{- $verifyErrMsg := "" -}}
```

Reply:
Once place we can do -- if required at multiple places. IMO, it just fine if we are passing at one place.
We will see later if required.

Reply:
Also -- it has some format specifier. Not sure how will it behave.
Will take it later.